client/main.lua

```diff
@@ -2099,39 +2099,20 @@ end)
 ------------------------------refreshskin-------------------
 
 RegisterCommand("refreshskin", function(source, args, rawCommand)
-
-
-    local playerPed = PlayerPedId()
-    local maxhealth = GetEntityMaxHealth(playerPed)
-    local health = GetEntityHealth(playerPed)
-
-    reloadSkin(health)
-
+    	local ped = PlayerPedId()
+	local model = GetEntityModel(ped)
+        RequestModel(model)
+	Wait(1000) -- Safety Waits
+    	SetPlayerModel(PlayerId(), model)
+    	SetModelAsNoLongerNeeded(model) 
+	Wait(1000) -- Safety Waits
+   	TriggerServerEvent("qb-clothes:loadPlayerSkin") -- LOADING PLAYER'S CLOTHES
+    	TriggerServerEvent("qb-clothing:loadPlayerSkin") -- LOADING PLAYER'S CLOTHES - Event 2
+	for k, v in pairs(GetGamePool('CObject')) do
+        	if IsEntityAttachedToEntity(ped, v) then
+            		SetEntityAsMissionEntity(v, true, true)
+            		DeleteObject(v)
+            		DeleteEntity(v)
+        	end
+    	end
 end)
-
-function reloadSkin(health)
-
-    local model = nil
-
-    local gender = QBCore.Functions.GetPlayerData().charinfo.gender
-
-    if gender == 1 then -- Gender is ONE for FEMALE
-    model = GetHashKey("mp_f_freemode_01") -- Female Model
-    else 
-    model = GetHashKey("mp_m_freemode_01") -- Male Model
-    end 
-
-    RequestModel(model)
-
-    SetPlayerModel(PlayerId(), model)
-    SetModelAsNoLongerNeeded(model)
-    Citizen.Wait(1000) -- Safety Delay
-    
-    TriggerServerEvent("qb-clothes:loadPlayerSkin") -- LOADING PLAYER'S CLOTHES
-    TriggerServerEvent("qb-clothing:loadPlayerSkin") -- LOADING PLAYER'S CLOTHES - Event 2
-
-    SetPedMaxHealth(PlayerId(), maxhealth)
-    Citizen.Wait(1000) -- Safety Delay
-    SetEntityHealth(PlayerPedId(), health)
```

Comment:
I noticed when testing that if I /refreshskin'd whilst injured it set my health to 100 again. Could you re-add the SetEntityHealth stuff please (in the event is fine)